tests/unit/pubsub_v1/subscriber/test_streaming_pull_manager.py

```diff
@@ -1130,3 +1357,176 @@ def test_activate_ordering_keys_stopped_scheduler():
     manager.activate_ordering_keys(["key1", "key2"])
 
     manager._messages_on_hold.activate_ordering_keys.assert_not_called()
+
+
+@mock.patch("grpc_status.rpc_status.from_call")
+@mock.patch("google.protobuf.any_pb2.Any.Unpack")
+def test_get_ack_errors_unable_to_unpack(from_call, unpack):
+    manager = make_manager()
+
+    st = status_pb2.Status()
+    st.code = code_pb2.Code.INTERNAL
+    st.message = "qmsg"
+    error_info = error_details_pb2.ErrorInfo()
+    error_info.metadata["ack_1"] = "error1"
+    st.details.add().Pack(error_info)
+    mock_gprc_call = mock.Mock(spec=grpc.Call)
+    exception = exceptions.InternalServerError("msg", errors=(), response=mock_gprc_call)
+    from_call.return_value = st
+    # Unpack() failed
+    unpack.return_value = None
+
+    assert not streaming_pull_manager._get_ack_errors(exception)
+
+@mock.patch("grpc_status.rpc_status.from_call")
+def test_get_ack_errors(from_call):
+    manager = make_manager()
+
+    # exception = exceptions.from_grpc_status(grpc.StatusCode.INTERNAL, "message")
+    exception = exceptions.InternalServerError("msg", errors=(), response=None)
+    # No response obj
+    assert not streaming_pull_manager._get_ack_errors(exception)
+
+    mock_gprc_call = mock.Mock(spec=grpc.Call)
+    exception = exceptions.InternalServerError("msg", errors=(), response=mock_gprc_call)
+    from_call.return_value = None
+    # rpc_status.from_call() returned None
+    assert not streaming_pull_manager._get_ack_errors(exception)
+
+    mock_gprc_call = mock.Mock(spec=grpc.Call)
+    exception = exceptions.InternalServerError("msg", errors=(), response=mock_gprc_call)
+    from_call.return_value = None
+    # rpc_status.from_call() returned None
+    assert not streaming_pull_manager._get_ack_errors(exception)
+
+    mock_gprc_call = mock.Mock(spec=grpc.Call)
+    exception = exceptions.InternalServerError("msg", errors=(), response=mock_gprc_call)
+    from_call.side_effect = ValueError("val error msg")
+    # ValueError thrown, so return None
+    assert not streaming_pull_manager._get_ack_errors(exception)
+
+    st = status_pb2.Status()
+    st.code = code_pb2.Code.INTERNAL
+    st.message = "qmsg"
+    error_info = error_details_pb2.ErrorInfo()
+    error_info.metadata["ack_1"] = "error1"
+    st.details.add().Pack(error_info)
+    mock_gprc_call = mock.Mock(spec=grpc.Call)
+    exception = exceptions.InternalServerError("msg", errors=(), response=mock_gprc_call)
+    from_call.side_effect = None
+    from_call.return_value = st
+    # happy case - errors returned in a map
+    ack_errors = streaming_pull_manager._get_ack_errors(exception)
+    assert ack_errors
+    assert ack_errors["ack_1"] == "error1"
+
+
+def test_process_errors():
+    # no requests so no items in results lists
```

Comment:
This unit test tests many different aspects of `_process_errors()`, let's split it into specialized test cases.

(the same can probably be done with `test_get_ack_errors()`)